### PR TITLE
Add ":" to alphabet for default salt for namespaced models

### PR DESCRIPTION
**Pull request by vitobotta**
Hi!

I have made a tiny change to include the colon character in the alphabet used to calculate the default salt. Without that, idy won't work with models like "Files::Image" that are wrapped into a module or other class. Instead, an error is returned when idy attempts to find the index of that character in the alphabet.

Thanks in advance!

### CODE REVIEW

**Comment by vitobotta** [comment]
I swear the pull requests list was empty when I checked! When Github redirected to the list after I created my PR, it then showed #9 . Sorry about that

**Comment by vitobotta** [comment]
I will add a spec so you may merge this since the other PR hasn't been updated.

**Comment by vitobotta** [comment]
Hi again, I have added a spec and changed the implementation. Because the default salt takes into account only the first N characters, I am discarding the namespace so to use the name of the class only. It doesn't make too much difference but I think it's better. Thanks

File: lib/idy/extension.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ def idy_decode(hash, salt: self.salt)
       def idy_default_salt
         alphabet = Array('a'..'z')
 
-        indexes = name.downcase.split('').map do |char|
+        indexes = name.split("::").last.downcase.split('').map do |char|
```
Comment (wbotelhos) [inline lib/idy/extension.rb:49]: What do you think about let just the `shift` restrict the size, not two side effects, and just ignoring the colon like:

```
safe_letters = name.downcase.split('') - [':']
indexes      = safe_letters.map { |char| alphabet.index(char) + 1 }
```

Comment (rafaeldev) [reply lib/idy/extension.rb:49]: I think if use the `#tr` method is more efficient. LIke this:
```suggestion
        indexes = name..tr(':', '').last.downcase.split.map do |char|
```

The idea here is to remove the `":"` character.

Comment (wbotelhos) [reply lib/idy/extension.rb:49]: @rafaeldev I agree, but in this case, we don't have the `last`, since we don't have the `split` anymore.

**Pull request merged**